Refactor(eos_designs): Remove EVPN limitation for underlay_filter_peer_as

emilarista opened the pull request:
## Change Summary

Removes the check for device evpn_role: client/server when deciding whether to configure the route-map for underlay_filter_peer_as. Also raises an exception when configuring this knob together with default VRF EVPN services since this will create conflicting config.

Created negative unit test and updated documentation.

## Component(s) name

`arista.avd.eos_designs`

## How to test
Tested with molecule and customer repo.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

emilarista committed:
working on negative unit test

emilarista committed:
update docs

ClausHolbechArista committed:
Adjust negative test and error message

ClausHolbechArista reviewed (changes requested):
I have updated the negative tests. You only need to add a new play when the failure happens in the eos_designs_facts phase.
You still need to remove the EVPN restriction from the underlay filter feature and add a positive molecule test for that combination.

emilarista committed:
added unit test for evpn/underlay_filter combo

emilarista commented:
> I have updated the negative tests. You only need to add a new play when the failure happens in the eos_designs_facts phase. You still need to remove the EVPN restriction from the underlay filter feature and add a positive molecule test for that combination.

I already removed the restriction, now added positive molecule test!

emilarista committed:
add second peer  to unit test

ClausHolbechArista committed:
Fix old hidden issue

ClausHolbechArista committed:
Update CI artifacts after rebase

ClausHolbechArista committed:
Merge branch 'devel' into refactor_underlay_filter

ClausHolbechArista committed:
Merge branch 'devel' into refactor_underlay_filter

carlbuchmann reviewed (approved):
LGTM tested the correct configuration with both RFC5549 and ipv4 underlay.
Tested negative test  with EVPN services in the default VRF
